test/router/path_trie_tail_test.dart

```diff
@@ -0,0 +1,265 @@
+import 'package:relic/src/router/normalized_path.dart';
+import 'package:relic/src/router/path_trie.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('PathTrie Tail (**) Matching', () {
+    late PathTrie<int> trie;
+
+    setUp(() {
+      trie = PathTrie<int>();
+    });
+
+    test(
+        'Given a trie with path /static/**, '
+        'when /static/css/style.css is looked up, '
+        'then it matches with correct value and remaining path', () {
+      trie.add(NormalizedPath('/static/**'), 1);
+      final result = trie.lookup(NormalizedPath('/static/css/style.css'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/static');
+      expect(result.remaining.path, '/css/style.css');
+    });
+
+    test(
+        'Given a trie with path /files/**, '
+        'when /files/a/b/c/doc.txt (multiple segments for tail) is looked up, '
+        'then it matches with correct value and remaining path', () {
+      trie.add(NormalizedPath('/files/**'), 1);
+      final result = trie.lookup(NormalizedPath('/files/a/b/c/doc.txt'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/files');
+      expect(result.remaining.path, '/a/b/c/doc.txt');
+    });
+
+    test(
+        'Given a trie with path /archive/**, '
+        'when /archive/ (ends where tail starts) is looked up, '
+        'then it matches with an empty remaining path', () {
+      trie.add(NormalizedPath('/archive/**'), 1);
+      final result = trie.lookup(NormalizedPath('/archive/'));
```

Comment:
~~What if this were just `/archive` (no trailing slash)?~~

Again confused that below is basically testing exactly this, but using another naming again.

Personally I would probably do better with parameterized test tables, but that might not be the convention for this project.

Reply:
Actually I do have a `parameterizedTest`  function in this project. I should use that here.